src/main/java/com/jcraft/jsch/Channel.java

```diff
@@ -45,38 +45,43 @@
 
   static int index=0; 
   private static Vector<Channel> pool=new Vector<>();
-  static Channel getChannel(String type){
+  static Channel getChannel(String type, Session session){
+    Channel ret = null;
     if(type.equals("session")){
-      return new ChannelSession();
+      ret = new ChannelSession();
```

Comment:
I wonder if we should perhaps we just add a `Session session` argument to each of the `ChannelXXX` implementations instead of adding an explicit `setSession(session)` call below?
That would then allow us to remove the `addChannel()` method in `Session` & `PortWatcher` classes I think?
@mwiede any thoughts?